modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/CommentParsingTest.java

```diff
@@ -0,0 +1,203 @@
+ * </ol>
+ *
+ * <p>According to SQL standard, SQL text containing one or more instances of comment is equivalent to the same SQL text with the comment
+ * replaced with newline.
+ */
+public class CommentParsingTest extends AbstractParserTest {
+    private static final IgniteLogger LOG = Loggers.forClass(CommentParsingTest.class);
+
+    private static final String NL = System.lineSeparator();
+
+    @ParameterizedTest
+    @EnumSource(Statement.class)
+    void leadingSimpleComment(Statement statement) {
+        String originalQueryString = statement.text;
+        String queryWithComment = "-- this is comment " + NL + originalQueryString;
+
+        assertQueries(
+                originalQueryString,
+                queryWithComment
+        );
+    }
+
+    @ParameterizedTest
+    @EnumSource(Statement.class)
+    void trailingSimpleComment(Statement statement) {
+        String originalQueryString = statement.text;
+        String queryWithComment = originalQueryString + NL + "-- this is comment";
+
+        assertQueries(
+                originalQueryString,
+                queryWithComment
+        );
+    }
+
+    @Test
+    void emptyStatementSimpleComment() {
+        assertThrowsSqlException(
+                Sql.STMT_PARSE_ERR,
+                "Failed to parse query",
+                () -> parse("-- this is comment")
+        );
+    }
+
+    /**
+     * This test injects simple comment before random line break.
+     */
+    @ParameterizedTest
+    @EnumSource(Statement.class)
+    void infixSimpleComment(Statement statement) {
+        int iterations = 50;
+        long seed = ThreadLocalRandom.current().nextLong();
+
+        LOG.info("Seed is {}", seed);
+
+        Random rnd = new Random(seed);
+
+        // it's well-known query that has less than
+        // Integer.MAX_VALUE lines
+        @SuppressWarnings("NumericCastThatLosesPrecision")
+        int linesCount = (int) statement.text.lines().count();
+
+        for (int i = 0; i < iterations; i++) {
+            int lineToInject = Integer.min(statement.maxLineBreak, rnd.nextInt(linesCount));
+
+            int lineNum = 0;
+            IgniteStringBuilder sb = new IgniteStringBuilder();
+            for (String line : statement.text.split(NL)) {
+                sb.app(line);
+
+                if (lineNum++ == lineToInject) {
+                    sb.app(" -- this is simple comment");
+                }
+
+                sb.app(NL);
+            }
+
+            assertQueries(statement.text, sb.toString());
+        }
+    }
+
+    private void assertQueries(String expected, String actual) {
+        SqlNode expectedAst;
+        SqlNode actualAst;
+        try {
+            expectedAst = parse(expected);
+        } catch (RuntimeException ex) {
+            System.err.println(expected);
+
+            throw ex;
+        }
+
+        try {
+            actualAst = parse(actual);
+        } catch (RuntimeException ex) {
+            System.err.println(expected);
```

Comment:
what reason to have these try/catch blocks with printing to system error?

Reply:
to provide context in case of error

Reply:
ok, understand. But why not `LOG.error(expected);` ?

Reply:
because test runner doesn't use our log infrastructure to print out exceptions ¯\_(ツ)_/¯ that way problematic query will be close to the original exception

Reply:
anyway, replaced `System.err` with `LOG.error`